packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -140,7 +423,65 @@ async function _getReferenceDocument(input: string): Promise<string> {
             const s3Client = new S3({ region })
             const resp = await s3Client.getObject({ Bucket: bucket, Key: key }).promise()
             return resp.Body!.toString()
-        } catch (e: any) {}
+        } catch (e: any) {
+            if (e.message.includes('Invalid S3 URI')) {
+                throw new PolicyChecksError('Invalid file path or S3 URI', PolicyChecksErrorCode.FileReadError)
+            } else {
+                throw new PolicyChecksError(e.message, PolicyChecksErrorCode.FileReadError)
+            }
+        }
     }
-    return ''
+}
+
+//Check if Cfn and Tf tools are installed
+function arePythonToolsInstalled(): boolean {
+    let cfnToolInstalled = true
+    let tfToolInstalled = true
+    try {
+        execSync('tf-policy-validator')
+    } catch (err: any) {
+        if (err.message.includes('command not found')) {
+            tfToolInstalled = false
+        }
+    }
+    try {
+        execSync('cfn-policy-validator')
+    } catch (err: any) {
+        if (err.message.includes('command not found')) {
+            cfnToolInstalled = false
+        }
+    }
+    return cfnToolInstalled && tfToolInstalled
+}
+
+// Since TypeScript can only get the CLI tool's error output as a string, we have to parse and sanitize it ourselves
+function getCliErrorMessage(message: string): string {
```

Comment:
is there no exit code/error code that we can more reliably depending on?

Reply:
The CLI error message varies and is not exactly consistent. As of now there are cracks of which the error output is simply the CLI error message, but we will revisit error messages. Adding to ticket.